Index: SPD-classes/src/main/java/com/watabou/utils/PlatformSupport.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SPD-classes/src/main/java/com/watabou/utils/PlatformSupport.java b/SPD-classes/src/main/java/com/watabou/utils/PlatformSupport.java
--- a/SPD-classes/src/main/java/com/watabou/utils/PlatformSupport.java	(revision f4f372585cae923b6cb363ec83eb4ffc54d34adb)
+++ b/SPD-classes/src/main/java/com/watabou/utils/PlatformSupport.java	(revision c096bf538a867d4e7fe9d63cb8dbb41403c646c8)
@@ -30,7 +30,12 @@

 import java.util.HashMap;

+import com.zrp200.scrollofdebug.PackageTrie;
+
 public abstract class PlatformSupport {
+	public PackageTrie findClasses(String pckgname) throws ClassNotFoundException {
+		return PackageTrie.getClassesForPackage(pckgname);
+	}

 	public abstract void updateDisplaySize();

Index: core/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java
===================================================================
diff --git a/core/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java b/SPD-classes/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java
rename from core/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java
rename to SPD-classes/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java
--- a/core/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java	(revision f4f372585cae923b6cb363ec83eb4ffc54d34adb)
+++ b/SPD-classes/src/main/java/com/zrp200/scrollofdebug/PackageTrie.java	(revision c096bf538a867d4e7fe9d63cb8dbb41403c646c8)
@@ -183,7 +183,7 @@
                         Class cls = Class.forName(pckgname + '.'
                                 + file.substring(0, file.length() - 6));
                         //if(canInstantiate(cls))
-                        trie.classes.add(cls);
+                            trie.classes.add(cls);
                     } catch (final NoClassDefFoundError e) {
                         // do nothing. this class hasn't been found by the
                         // loader, and we don't care.
Index: android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java b/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java
--- a/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java	(revision f4f372585cae923b6cb363ec83eb4ffc54d34adb)
+++ b/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java	(revision ae666820a450755751e1e98a7638c87362402e95)
@@ -41,12 +41,41 @@
 import com.watabou.noosa.Game;
 import com.watabou.utils.PlatformSupport;

+import java.io.IOException;
+import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;

+import dalvik.system.DexFile;
+import com.zrp200.scrollofdebug.PackageTrie;
+
 public class AndroidPlatformSupport extends PlatformSupport {
-
+
+	@Override
+	public PackageTrie findClasses(String pkgName) throws ClassNotFoundException {
+		return new PackageTrie() {
+			{
+				try {
+					Enumeration<String> entries = new DexFile(AndroidLauncher.instance
+							.getContext()
+							.getPackageCodePath()
+					).entries();
+					String n; while(entries.hasMoreElements()) {
+						n = entries.nextElement();
+						if(n.contains(pkgName)) try {
+							addClass(Class.forName(n), pkgName);
+						} catch (Exception e) {
+							e.printStackTrace();
+						}
+					}
+				} catch (IOException e) {
+					//e.printStackTrace();
+				}
+			}
+		};
+	}
+
 	public void updateDisplaySize(){
 		if (SPDSettings.landscape() != null) {
 			AndroidLauncher.instance.setRequestedOrientation( SPDSettings.landscape() ?
Index: core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java b/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java
--- a/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java	(revision f4f372585cae923b6cb363ec83eb4ffc54d34adb)
+++ b/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java	(revision c096bf538a867d4e7fe9d63cb8dbb41403c646c8)
@@ -693,7 +693,7 @@
     public static PackageTrie trie = null; // loaded when needed.
     static {
         try {
-            trie = PackageTrie.getClassesForPackage(ROOT);
+            trie = ShatteredPixelDungeon.platform.findClasses(ROOT);
         } catch (ClassNotFoundException e) { ShatteredPixelDungeon.reportException(e); }
     }

Index: core/src/main/java/com/zrp200/scrollofdebug/v1_3_Compatibility_fixes_Android.patch
===================================================================
diff --git a/core/src/main/java/com/zrp200/scrollofdebug/v1_3_Compatibility_fixes_Android.patch b/core/src/main/java/com/zrp200/scrollofdebug/v1_3_Compatibility_fixes_Android.patch
new file mode 100644
--- /dev/null	(revision ae666820a450755751e1e98a7638c87362402e95)
+++ b/core/src/main/java/com/zrp200/scrollofdebug/v1_3_Compatibility_fixes_Android.patch	(revision ae666820a450755751e1e98a7638c87362402e95)
@@ -0,0 +1,43 @@
+Index: android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java b/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java
+--- a/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java	(revision 37eb579247b6694b3c80bacf589fadaf6de414b0)
++++ b/android/src/main/java/com/shatteredpixel/shatteredpixeldungeon/android/AndroidPlatformSupport.java	(date 1658003330856)
+@@ -57,7 +57,7 @@
+ 		return new PackageTrie() {
+ 			{
+ 				try {
+-					Enumeration<String> entries = new DexFile(AndroidGame.instance
++					Enumeration<String> entries = new DexFile(AndroidLauncher.instance
+ 							.getContext()
+ 							.getPackageCodePath()
+ 					).entries();
+Index: core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java b/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java
+--- a/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java	(revision 37eb579247b6694b3c80bacf589fadaf6de414b0)
++++ b/core/src/main/java/com/zrp200/scrollofdebug/ScrollOfDebug.java	(date 1658189051636)
+@@ -36,7 +36,7 @@
+ import com.shatteredpixel.shatteredpixeldungeon.ui.ScrollPane;
+ import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
+ // WndTextInput (added in v0.9.4)
+-import com.shatteredpixel.shatteredpixeldungeon.ui.WndTextInput;
++import com.shatteredpixel.shatteredpixeldungeon.windows.WndTextInput;
+ // Output
+ import com.shatteredpixel.shatteredpixeldungeon.utils.GLog;
+
+@@ -131,7 +131,7 @@
+     @Override
+     public void doRead() {
+         collect(); // you don't lose scroll of debug.
+-        GameScene.show(new WndTextInput("Enter Command:", "", 100, false,
++        GameScene.show(new WndTextInput("Enter Command:", null, "", 100, false,
+                 "Execute", "Cancel") {
+             @Override public void onSelect(boolean positive, String text) {
+                 if(!positive) return;
